pkg/components/aws-ebs-csi-driver/component_test.go

```diff
@@ -137,3 +139,134 @@ func TestStorageClassDisabled(t *testing.T) {
 		t.Fatalf("Default storage class should not be set")
 	}
 }
+
+func TestConversion(t *testing.T) { //nolint:funlen
+	testCases := []struct {
+		name                 string
+		inputConfig          string
+		expectedManifestName string
+		expected             string
+		jsonPath             string
+	}{
+		{
+			name:                 "no_tolerations_node",
+			inputConfig:          `component "aws-ebs-csi-driver" {}`,
+			expectedManifestName: "aws-ebs-csi-driver/templates/node.yaml",
+			jsonPath:             "{.spec.template.spec.tolerations[0].operator}",
+			expected:             "Exists",
+		},
+		{
+			name: "tolerations_node",
+			inputConfig: `component "aws-ebs-csi-driver" {
+							tolerations {
+								key      = "lokomotive.io"
+								operator = "Equal"
+								value    = "awesome"
+								effect   = "NoSchedule"
+							}
+						}`,
+			expectedManifestName: "aws-ebs-csi-driver/templates/node.yaml",
+			jsonPath:             "{.spec.template.spec.tolerations[0].value}",
+			expected:             "awesome",
+		},
+		{
+			name:                 "no_tolerations_csi_controller",
+			inputConfig:          `component "aws-ebs-csi-driver" {}`,
+			expectedManifestName: "aws-ebs-csi-driver/templates/controller.yaml",
+			jsonPath:             "{.spec.template.spec.tolerations[0].operator}",
+			expected:             "Exists",
+		},
+		{
+			name: "tolerations_csi_controller",
+			inputConfig: `component "aws-ebs-csi-driver" {
+							tolerations {
+								key      = "lokomotive.io"
+								operator = "Equal"
+								value    = "awesome"
+								effect   = "NoSchedule"
+							}
+						}`,
```

Comment:
Indentation is interesting in those test cases :smile: But I'm not sure what would be the right way to format it.